cli/cmd/plugin/ui/web/tanzu-ui/src/views/management-cluster/vsphere/vsphere-mc-common/VsphereCredentialsStep.tsx

```diff
@@ -183,26 +186,32 @@ export function VsphereCredentialsStep(props: Partial<StepProps>) {
     };
 
     const login = () => {
+        setConnecting(true);
         const vSphereCredentials = {
             username: vsphereState[STORE_SECTION_FORM][VSPHERE_FIELDS.USERNAME],
             host: vsphereState[STORE_SECTION_FORM][VSPHERE_FIELDS.SERVERNAME],
             password: vsphereState[STORE_SECTION_FORM][VSPHERE_FIELDS.PASSWORD],
             thumbprint: useThumbprint ? thumbprint : '',
             insecure: !useThumbprint,
         } as VSphereCredentials;
-        VsphereService.setVSphereEndpoint(vSphereCredentials).then(
-            () => {
-                setConnection(true);
-            },
-            (reason: any) => {
-                console.warn(`When trying to connect to the server, encountered error ${JSON.stringify(reason)}`);
-                let msg = 'Unable to connect to server! (See console for details.)';
-                if (reason?.status === SERVER_RESPONSE_BAD_CREDENTIALS) {
-                    msg = 'Incorrect username/password combination! Please try again. (See console for technical details.)';
+        // TODO: remove setTimeout(), which is just here to simulate a backend call delay
+        setTimeout(() => {
+            VsphereService.setVSphereEndpoint(vSphereCredentials).then(
```

Comment:
Adding to my previous comment on having config. we could have a custom hook that returns the connection state. so the state is managed in custom hook and not in the implementation components.

Reply:
And please use async await to call the api